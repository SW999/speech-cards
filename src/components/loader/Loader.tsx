import React, { FunctionComponent } from 'react';
import '../../scss/components/_loader.scss';

type LoaderProps = {
  onlyText?: boolean;
  placeholder?: string;
};

const Loader: FunctionComponent<LoaderProps> = ({
  onlyText = false,
  placeholder = 'Loading...',
}) => {
  return (
    <div className="loader-wrapper">
      {onlyText ? (
        <h1>{placeholder}</h1>
      ) : (
        <svg
          className="loader"
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 320 320"
        >
          <circle cx="160" cy="160" r="140" stroke="#BD4E31" />
          <circle cx="160" cy="160" r="115" stroke="#f6ce4b" />
          <circle cx="160" cy="160" r="90" stroke="#1998d3" />
          <circle cx="160" cy="160" r="65" stroke="#2b2b2b" />
        </svg>
      )}
    </div>
  );
};

export default Loader;
