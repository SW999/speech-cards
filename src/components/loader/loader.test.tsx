import React from 'react';
import { render } from '@testing-library/react';
import Loader from './Loader';

describe('Loader component', () => {
  it('Loader should take a snapshot with spinner', () => {
    const { asFragment } = render(<Loader />);

    expect(asFragment()).toMatchSnapshot('Loader with spinner');
  });

  it('Loader should take a snapshot with text', () => {
    const { asFragment } = render(<Loader onlyText />);

    expect(asFragment()).toMatchSnapshot('Loader with text');
  });
});
