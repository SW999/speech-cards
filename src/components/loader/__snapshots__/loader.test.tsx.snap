// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Loader should take a snapshot with spinner: Loader with spinner 1`] = `
<DocumentFragment>
  <div
    class="loader-wrapper"
  >
    <svg
      class="loader"
      viewBox="0 0 320 320"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="160"
        cy="160"
        r="140"
        stroke="#BD4E31"
      />
      <circle
        cx="160"
        cy="160"
        r="115"
        stroke="#f6ce4b"
      />
      <circle
        cx="160"
        cy="160"
        r="90"
        stroke="#1998d3"
      />
      <circle
        cx="160"
        cy="160"
        r="65"
        stroke="#2b2b2b"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`Loader component Loader should take a snapshot with text: Loader with text 1`] = `
<DocumentFragment>
  <div
    class="loader-wrapper"
  >
    <h1>
      Loading...
    </h1>
  </div>
</DocumentFragment>
`;
